### ISSUE TITLE
Use ttf-parser for font parsing

### ISSUE THREAD

**Issue by notgull** [dfrg/swash#36]
I see that this crate uses its own code for TTF parsing. Is there a reason why `ttf-parser` isn't used here? It has no dependencies and is written in exclusively safe code.

**Comment by dfrg** [comment]
The primary value proposition of swash is support for hinting which is not possible to implement on top of ttf-parser without significant architectural changes to that crate.

**Comment by dfrg** [comment]
Rustybuzz implements [shaping](https://en.m.wikipedia.org/wiki/Complex_text_layout). AFAIK swash is the only rust library with full support for [hinting](https://en.m.wikipedia.org/wiki/Font_hinting).

**Comment by Maximkaaa** [comment]
Yeah, sorry, I completely misunderstood things. The more I understand about text rendering the more I understand that I don't understand more than I understand...

The source of my confusion was that I saw a reference to `ttf-parser` [HintDevice](https://docs.rs/ttf-parser/latest/ttf_parser/gpos/struct.HintingDevice.html) and thought they use it (though they clearly don't since they don't do rasterizing).

**Comment by dfrg** [comment]
Yes, text rendering is a very deep rabbit hole :)
